aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -466,6 +534,27 @@ void flip_kernel(TensorIterator& iter, const bool quantized) {
         });
     });
   } else {
+    // Special case: horizontal flip with vectorization and input is contiguous
+    // Context: horizontal flip leads to strides[0] < 0 and
+    // thus is_contiguous condition is not satisfied and non-vectorized code path is taken.
+    auto output_strides = iter.strides(0);
+    auto input_strides = iter.strides(1);
+    if (iter.ndim() > 1 && output_strides[0] < 0 && input_strides[0] == iter.element_size(1)) {
```

Comment:
nit: `iter.ndim()==1` should benefit from this too?

Reply:
Indeed, flipping 1D tensors can be faster using this code path. Thanks!